agent/package.rb

```diff
@@ -117,13 +121,14 @@ def self.do_pkg_action(package, action, reply, version=nil)
         provider = Package.load_provider_class(Package.package_provider).new(package, Package.provider_options(Package.package_provider, version))
         result = provider.send(action)
 
+        # somewhere around here this is converting the hash to a string :\
         if action == :status
           result.each do |k,v|
-            reply[k] = v.to_s
+            reply[k] = v.is_a?(Array) || v.is_a?(Hash) ? v : v.to_s
```

Comment:
@ripienaar this was tripping up the rendering of the output - if the hash was more than 1 level deep it would render the keys as strings.

This .. seems to do what I want though?

```json
  {
    "agent": "package",
    "action": "search",
    "sender": "testhost",
    "statuscode": 5,
    "statusmsg": "1 packages found",
    "data": {
      "available_packages": [
        {
          "arch": "x86_64",
          "ensure": "1.4.8-1.el6",
          "epoch": "0",
          "name": "GeoIP",
          "provider": "yum",
          "release": "1.el6",
          "version": "1.4.8"
        }
      ]
    }
  }
]
```

Testing remove/install/search/status seems to work as expected now, at least for the yum provider.

Update is looking a bit weird.  The output is kinda inconsistent.